SpatialGDK/Source/SpatialGDK/Private/Interop/WorkingSetsCommon.cpp

```diff
@@ -0,0 +1,121 @@
+
+#include "Interop/WorkingSetsCommon.h"
+
+#include "SpatialView/EntityComponentTypes.h"
+#include "SpatialView/EntityDelta.h"
+#include "SpatialView/EntityView.h"
+
+#include "Schema/ActorGroupMember.h"
+#include "Schema/AuthorityIntent.h"
+#include "Schema/Interest.h"
+#include "Schema/StandardLibrary.h"
+#include "SpatialView/ComponentData.h"
+#include "SpatialView/ViewCoordinator.h"
+
+namespace SpatialGDK
+{
+FWorkingSetCommonData::FWorkingSetCommonData(const EntityViewElement& Entity)
+	: Request(*Entity.Components.FindByPredicate(ComponentIdEquality{ FActorSetRequest::ComponentId }))
+	, Response(*Entity.Components.FindByPredicate(ComponentIdEquality{ FActorSetResponse::ComponentId }))
+{
+}
+
+void FWorkingSetCommonData::Update(const EntityDelta& Delta)
+{
+	if (!ensureMsgf(Delta.Type == EntityDelta::UPDATE, TEXT("FWorkingSet::Update must be called with UPDATE entity deltas only!")))
+	{
+		return;
+	}
+	for (const ComponentChange& Change : Delta.ComponentUpdates)
+	{
+		const Schema_Object* UpdatedObject = nullptr;
+		switch (Change.Type)
+		{
+		case ComponentChange::UPDATE:
+			UpdatedObject = Schema_GetComponentUpdateFields(Change.Update);
+			break;
+		case ComponentChange::COMPLETE_UPDATE:
+			UpdatedObject = Schema_GetComponentDataFields(Change.CompleteUpdate.Data);
+			break;
+		default:
+			unimplemented();
+		}
+		if (!ensure(UpdatedObject != nullptr))
+		{
+			continue;
+		}
+		if (Change.ComponentId == FActorSetResponse::ComponentId)
+		{
+			Response.ApplySchema(*UpdatedObject);
+		}
+		else if (Change.ComponentId == FActorSetRequest::ComponentId)
+		{
+			Request.ApplySchema(*UpdatedObject);
+		}
+	}
+}
+
+TArray<ComponentData> CreateWorkingSetComponents(const Worker_PartitionId AuthServerWorkerPartitionId, const FActorSetRequest& Request)
+{
+	TArray<ComponentData> Components;
+
+	Components.Emplace(Request.CreateComponentData());
+	Components.Emplace(FActorSetResponse().CreateComponentData());
+
+	auto Convert = [](Worker_ComponentData Component) {
+		ComponentData Converted(OwningComponentDataPtr(Component.schema_type), Component.component_id);
+		return Converted;
+	};
+
+	{
+		Components.Emplace(Convert(Metadata(TEXT("WorkingSetMarker")).CreateComponentData()));
+	}
+
+	{
+		// Give the auth UnrealWorker authority over the Request component, and the strategy worker auth over the Response component
+		AuthorityDelegation Delegation;
+		Delegation.Delegations.Emplace(SpatialConstants::STRATEGY_WORKER_AUTH_COMPONENT_SET_ID,
+									   SpatialConstants::INITIAL_STRATEGY_PARTITION_ENTITY_ID);
+		Delegation.Delegations.Emplace(SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID, AuthServerWorkerPartitionId);
+
+		Components.Emplace(Convert(Delegation.CreateComponentData()));
+	}
+	{
+		Interest In;
+		Query Query;
+		QueryConstraint SelfConstraint;
+		SelfConstraint.bSelfConstraint = true;
+		Query.Constraint = SelfConstraint;
+		Query.ResultComponentIds = {
+			FActorSetRequest::ComponentId,
```

Comment:
Consider only specifying interest in Request on StrategyWorker, and Response on ServerWorker.